pkg/configuration/configuration.go

```diff
@@ -1,72 +1,93 @@
 package configuration
 
 import (
-	app "github.com/scribd/go-sdk/pkg/app"
-	database "github.com/scribd/go-sdk/pkg/database"
-	instrumentation "github.com/scribd/go-sdk/pkg/instrumentation"
-	logger "github.com/scribd/go-sdk/pkg/logger"
-	"github.com/scribd/go-sdk/pkg/pubsub"
-	server "github.com/scribd/go-sdk/pkg/server"
-	tracking "github.com/scribd/go-sdk/pkg/tracking"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/scribd/go-sdk/pkg/configuration/apps"
+	"github.com/scribd/go-sdk/pkg/configuration/builder"
 )
 
-// Config is an app-wide configuration
-type Config struct {
-	App             *app.Config
-	Database        *database.Config
-	Instrumentation *instrumentation.Config
-	Logger          *logger.Config
-	Server          *server.Config
-	Tracking        *tracking.Config
-	PubSub          *pubsub.Config
+const (
+	defaultEnv       = "development"
+	defaultConfigDir = "config"
+
+	appRootEnvKey = "APP_ROOT"
+	appNameEnvKey = "APP_NAME"
+	envEnvKey     = "APP_ENV"
+)
+
+// Configuration is an app-wide configuration
+type Configuration struct {
+	AppName string
+	AppEnv  string
+	AppRoot string
+
+	Database        apps.Database
+	Server          apps.Server
+	Instrumentation apps.Instrumentation
+	Logger          apps.Logger
+	Tracking        apps.Tracking
+	PubSub          apps.PubSub
 }
 
-// NewConfig returns a new Config instance
-func NewConfig() (*Config, error) {
-	config := &Config{}
+// NewConfig returns a new Config instance in default location.
+func NewConfig(configs ...Configurable) (*Configuration, error) {
+	return NewConfigWithPath(defaultConfigDir, configs...)
+}
 
-	appConfig, err := app.NewDefaultConfig()
-	if err != nil {
-		return config, err
-	}
+// NewConfigWithPath returns a new Config instance within a directory.
+// configDir is relative path of config directory.
+func NewConfigWithPath(confDir string, configurables ...Configurable) (*Configuration, error) {
+	appName := getAppNameMust()
+	appRoot := getAppRootMust()
+	appEnv := getEnvMust()
```

Comment:
app name and env was used by many app configs using `os.Getenv`, this way we can have them all set together and also can put some validation on top of it. now why did I fatal instead of returning error, is because I believe this error is different than others in 2 sense. first is that this is the first line of initializing configuration which is the first line of your main func, one of very first needs of program so make sense to fatal without need to see the log or trace (considering the is no logger too). the other is that this error is not coding related, so must be plain text that anyone from monitoring/devops team could get to it fast and easy.
what do you think?